### PR TITLE
Add docker hub publish workflow

### PR DESCRIPTION
**Pull request by emelvarol-cognite**

### CODE REVIEW

File: .github/workflows/publish_on_dockerhub.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    tags:
+      - 'release/*'
+
+name: Release
+defaults:
+  run:
+    shell: pwsh
+
+jobs:
+  create_release:
```
Comment (mathialo) [inline .github/workflows/publish_on_dockerhub.yml:12]: As far as I can see this doesn't create a github release? Why is this included? 🙂

**Review by mathialo — COMMENTED** [review]
In the release workflow, new versions are released to pypi whenever the pyproject.toml version is updated. Here we are doing a manual tagging, which I suspect we will often forget to do. Could we trigger this on merges to main, and only actually publish if the version number has changed?

File: .github/workflows/publish_on_dockerhub.yml (hunk 1 above)
Comment (mathialo) [reply .github/workflows/publish_on_dockerhub.yml:12]: btw, I think having github releases as well is a really good idea :)

**Comment by emelvarol-cognite** [comment]
> In the release workflow, new versions are released to pypi whenever the pyproject.toml version is updated. Here we are doing a manual tagging, which I suspect we will often forget to do. Could we trigger this on merges to main, and only actually publish if the version number has changed?

makes sense I'll check it out now

File: .github/workflows/publish_on_dockerhub.yml (hunk 1 above)
Comment (emelvarol-cognite) [reply .github/workflows/publish_on_dockerhub.yml:12]: this was just for version parsing, isn't it necessary?

**Comment by emelvarol-cognite** [comment]
@mathialo I suggest you unsubscribe from this one, I am trying and failing things to extract python app version so there will be a lot of spams. I can notify you when I manage things.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -50,4 +50,16 @@ jobs:
         TWINE_USERNAME: __token__
         TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
       run: twine upload --verbose dist/* || echo 'Version exists'
-      
\ No newline at end of file
+
+    - id: parse_version
+      name: "Parse version"
+      run: |
+        echo "::set-output name=version::$(PYTHONPATH=cognite/transformations_cli poetry run python -m cognite.transformations_cli --version | cut -c 30-)"
```
Comment (emelvarol-cognite) [inline .github/workflows/release.yml:57]: `transformations_cli, version 1.2.11` is printed, we cut `transformations_cli, version ` from that.

**Pull request closed without merging**